### PR TITLE
Move GCP related resources to separate dir

### LINKED ISSUE #600: Reorganize examples and integration tests so that it is organized by cloud
Our examples and integration tests should be organized in the following directory structure:

- `examples/<cloud>/<example>`
- `tests/<cloud>/<test>`

This allows us to break up the test routines so that we run each cloud separately. See https://github.com/gruntwork-io/terratest/pull/597 for example, where we broke out the `azure` examples and tests so that they run in a separate pipeline in CI.

The initial proposed list of clouds for `examples` and `tests` are:

- `aws`
- `gcp`
- `azure`
- `k8s`
- `oci`
- `local`

### PR DESCRIPTION
**Pull request by StupidScience**
Related to issue #600

### CODE REVIEW

**Comment by robmorgan** [comment]
I'm in favor of this, but I guess we should also add [build tags](https://github.com/gruntwork-io/terratest/blob/master/test/azure/terraform_azure_actiongroup_example_test.go#L1) similar to how we handle Azure in order to future proof it. @yorinasub17 thoughts?

@StupidScience did the tests pass correctly?

**Comment by StupidScience** [comment]
@robmorgan I've moved default packer variables to separate dir and now it works fine.
Also `// +build gcp` tag added

**Comment by brikis98** [comment]
I'm in favor of this 👍 

Note that this only partially fixes #600, so @StupidScience, could you rephrase your PR description so that issue isn't auto-closed when this PR is merged?

@robmorgan, I'll let you take it from here, run tests, etc.

**Comment by StupidScience** [comment]
@brikis98 what should I do to prevent issue autoclosing? Just remove issue reference?

**Comment by brikis98** [comment]
Yup! Or remove the word "fixes" as I think that's one of the keywords GitHub looks for in PR descriptions.

**Comment by StupidScience** [comment]
@brikis98 done

**Comment by brikis98** [comment]
Thanks!

**Comment by yorinasub17** [comment]
This LGTM! However, before we can kick off our build, I think the circleci config will need some surgery to make sure it runs the `gcp` packages.

**Comment by james00012** [comment]
Thanks for the contribution, and apologies for the long delay. This PR has been open for over a year without movement, so we're closing it as stale. Please feel free to reopen if you find time to pick it back up again. We appreciate the effort.
